test/e2e/network/dual_stack.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	apps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2edeploy "k8s.io/kubernetes/test/e2e/framework/deployment"
+
+	"net"
+)
+
+const (
+	// nginx image with ipv6
+	nginxImageWithIPv6 = "aramase/nginx-ipv6"
+	// network utils docker image
+	networkUtilImage = "aramase/nettools"
+)
+
+// Tests for ipv6 dual stack feature
+var _ = SIGDescribe("[Feature:IPv6DualStackAlphaFeature] [LinuxOnly]", func() {
+	f := framework.NewDefaultFramework("dualstack")
+
+	var cs clientset.Interface
+	var podClient *framework.PodClient
+
+	ginkgo.BeforeEach(func() {
+		// This test suite requires aks-engine cluster.
+		//framework.SkipUnlessProviderIs("azure")
+		cs = f.ClientSet
+		podClient = f.PodClient()
+	})
+
+	ginkgo.It("should have ipv4 and ipv6 internal node ip", func() {
+		// TODO (aramase) can switch to new function to get all nodes
+		nodeList := framework.GetReadySchedulableNodesOrDie(cs)
+
+		for _, node := range nodeList.Items {
+			// get all internal ips for node
+			internalIPs, err := framework.GetNodeInternalIPs(&node)
+			framework.ExpectNoError(err)
+
+			gomega.Expect(len(internalIPs)).Should(gomega.BeNumerically("==", 2))
+			// assert one is ipv4 and other is ipv6
+			gomega.Expect(isIPv4(internalIPs[0])).Should(gomega.BeTrue())
+			gomega.Expect(isIPv6(internalIPs[1])).Should(gomega.BeTrue())
+		}
+	})
+
+	ginkgo.It("should have ipv4 and ipv6 node podCIDRs", func() {
+		// TODO (aramase) can switch to new function to get all nodes
+		nodeList := framework.GetReadySchedulableNodesOrDie(cs)
+
+		for _, node := range nodeList.Items {
+			gomega.Expect(len(node.Spec.PodCIDRs)).Should(gomega.BeNumerically("==", 2))
+			// assert podCIDR is same as podCIDRs[0]
+			gomega.Expect(node.Spec.PodCIDR).Should(gomega.BeEquivalentTo(node.Spec.PodCIDRs[0]))
+			// assert one is ipv4 and other is ipv6
+			gomega.Expect(isIPv4CIDR(node.Spec.PodCIDRs[0])).Should(gomega.BeTrue())
```

Comment:
same as the previous comment, if we can support both cases, v4v6 and v6v4, I think that we should check only that both addresses belong to different families, something like
gomega.Expect(isIPv4(node.Spec.PodCIDRs[0]) != isIPv4(node.Spec.PodCIDRs[1])).Should(gomega.BeTrue())